paddle/fluid/framework/tensor_util.cc

```diff
@@ -43,20 +43,27 @@ void TensorCopy(const Tensor& src, const platform::Place& dst_place,
 
   dst->Resize(src.dims());
   dst->set_layout(src.layout());
+  auto src_place = src.place();
+  auto src_ptr = src.data<void>();
 #ifdef PADDLE_WITH_MKLDNN
   dst->set_format(src.format());
 #endif
-  auto src_place = src.place();
-  auto src_ptr = src.data<void>();
-  auto dst_ptr = dst->mutable_data(dst_place, src.type());
+  // oneDNN tensors due to padding may be of bigger size
+  // than numel()*size(type())
+  auto dst_ptr =
```

Comment:
Maybe the new block of code should be guarded with: `#ifdef PADDLE_WITH_MKLDNN` ?

Reply:
ok